### PR TITLE
aws_servicecatalog_portfolio_product_association

### PR DESCRIPTION
**Pull request by soumyo13**
Signed-off-by: Soumyodeep Karmakar <soumyo.k13@gmail.com>

### Description

It contains all the singular and plural resources of the above resource.

### Issues Resolved
This is a new resource. There are no issues to be fixed.

### Check List
Please fill box or appropriate ([x]) or mark N/A.
- [x] New functionality includes integration tests/controls
- [x] New Terraform resources
- [x] Documentation provided or updated for resources 
- [x] All Integration Tests pass
- [x] All Unit Tests pass
- [x] `rake lint` passes
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: libraries/aws_servicecatalog_portfolio_product_associations.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+             .register_column(:provider_names,                     field: :provider_name)
+             .install_filter_methods_on_resource(self, :table)
+
+  def initialize(opts = {})
+    super(opts)
+    validate_parameters(required: %i(product_id))
+    @query_params = {}
+    @query_params[:product_id] = opts[:product_id]
+    raise ArgumentError, "#{@__resource_name__}: product_id must be provided" unless opts[:product_id] && !opts[:product_id].empty?
+    @query_params[:product_id] = opts[:product_id]
```
Comment (Nirbhay1997) [inline libraries/aws_servicecatalog_portfolio_product_associations.rb:32]: is this correct  ?

**Review by IanMadd — APPROVED** [review]

File: libraries/aws_servicecatalog_portfolio_product_associations.rb (hunk 1 above)
Comment (soumyo13) [reply libraries/aws_servicecatalog_portfolio_product_associations.rb]: Changed

File: libraries/aws_servicecatalog_portfolio_product_associations.rb [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      @api_response.portfolio_details.each do |resp|
+        rows += [{ id: resp.id,
+                   arn: resp.arn,
+                   display_name: resp.display_name,
+                   description: resp.description,
+                   created_time: resp.created_time,
+                   provider_name: resp.provider_name }]
+      end
+      break unless @api_response.next_page_token
+      @query_params[:next_page_token] = @api_response.next_page_token
```
Comment (Nirbhay1997) [inline libraries/aws_servicecatalog_portfolio_product_associations.rb:52]: @query_params[:page_token]

**Review by Nirbhay1997 — APPROVED** [review]

**Review by Nirbhay1997 — APPROVED** [review]
LGTM

File: libraries/aws_servicecatalog_portfolio_product_associations.rb (hunk 2 above)
Comment (soumyo13) [reply libraries/aws_servicecatalog_portfolio_product_associations.rb:52]: done

**Pull request merged**